### DISCUSSION TITLE
fs5i nordick track  more realistic ride when running with Rouvy or Zwift

### DISCUSSION THREAD

**Discussion by gewahr** [cagnulein/qdomyos-zwift#1345] [General]
Hi Roberto. Question for you when you have time . When i connect my Elliptical nordictrack fs5i to QZ and then to zwift or Rouvy for running , is there a more realistic way to control the speed . Sometimes I accelerate and the speed in zwift doesn't seem correlated it would give  spped of 4.5/km then 6.5  in the same 2 seconds while i maintain the same cadence . It's like my treadmill is not sending the data all the time to Zwift or Rouvy. Also increasing the resistance of the treadmill does not have influence on any data in zwift or Rouvy. Sending you the debug file of march 6. No hurries. thanks
[debug-lun__mars_6_09_06_02_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/10909018/debug-lun__mars_6_09_06_02_2023.log)

**Comment by cagnulein** [comment]
@gewahr qz is calculating wattage based on speed and inclination for your elliptical (so that's why changing the resistance doesn't affect the speed in the game). Are you using zwift or rouvy in bike mode or run mode?

**Comment by gewahr** [comment]
Hi Roberto. thanks for your quick answer. I am using zwift and Rouvy in run mode with the elliptical . There is no inclination in this elliptical . It is just like a stairmaster ... I just find that if I increase the cadence let say from 50 to 86  (same as when i run faster ) , I should go faster ( so less minutes/km) and  my wattage should increase it is not always the case . And also even if the resistance is not affecting speed is there a way that QZ would  calculate wattage considering the extra effort that I am doing when I increase my resistance manually from 1 to let say 18 ?  basically increasing the resistance is the same as increasing the inclination in that elliptical it makes it harder to maintain the same cadence or speed.
If it is too complicated do not worry ...I'm trying to simulate normal running  which maybe could not be reproducible with this elliptical. Thanks.

**Reply by cagnulein** [reply]
@gewahr zwift and rouvy in run mode don't read the wattage, but only the speed. So a couple of questions:
1. when you increase the cadence, don't you see in QZ the speed changing? and on your elliptical screen?
2. we can create an algorithm to change the speed based on resistance in case, with a gain factor or something. It will be good to create a ticket for this.
3. same for the wattage of course, i already did this for #1267

**Comment by gewahr** [comment]
ok sure thanks. i will test it this week and come back with data for a possible algorithm. the machine looks like that
![NordicTrack-FreeStride-Trainer-FS5i-1-460x409](https://user-images.githubusercontent.com/98962725/223706010-c21ebfec-8831-4b57-abda-f2608de3b32c.png)

**Comment by cagnulein** [comment]
great!

Il giorno mer 8 mar 2023 alle 12:38 gewahr ***@***.***> ha
scritto:

> ok sure thanks. i will test it this week and come back with data for a
> possible algorithm
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/1345#discussioncomment-5240695>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWAMEQ4QV7AEPJF5SP3W3BVTFANCNFSM6AAAAAAVSL5VJM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Comment by gewahr** [comment]
hi Roberto i tested the elliptical with ifit not sure how i can get data that is usable. Sent you videos and pictures through a link from Smash ( like we transfert)  because it was too big to send by Email
Joined is additionnal data
[2023_03_08_20_03_Santa_Maddalena_Church_Walk,_Italy.csv](https://github.com/cagnulein/qdomyos-zwift/files/10962084/2023_03_08_20_03_Santa_Maddalena_Church_Walk._Italy.csv)
[2023_03_13_16_03_Manual_Workout.csv](https://github.com/cagnulein/qdomyos-zwift/files/10962085/2023_03_13_16_03_Manual_Workout.csv)
those do not show the resistance.
took manualley a few measurement 
[fs5i manual measurement.xlsx](https://github.com/cagnulein/qdomyos-zwift/files/10962195/fs5i.manual.measurement.xlsx)

Let me know if we can use any data

**Reply by cagnulein** [reply]
sorry for the delay @gewahr 

so for example picking up a 64 cadence value

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/violarob/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/violarob/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



resistance | speed mph | cadence rpm | power watts
-- | -- | -- | --
1 | 6.6 | 64 | 204
3 | 6.1 | 64 | 220
7 | 6.3 | 64 | 289
11 | 6.1 | 64 | 409
15 | 6.4 | 64 | 570
13 | 5.7 | 64 | 461



</body>

</html>

I don't know why the speed is not increasing (but I guess we can ignore this since we are already reading the actual speed value from the elliptical itself).

So I can linearize a power algorithm

![image](https://user-images.githubusercontent.com/48216/228259681-c8af41fa-9545-4a74-a5f6-c8d680544979.png)

what do you think? than i can also apply the same concept for the cadence. I will check also in your first debug log if the power value maybe it's already coming through
Let me know

**Comment by gewahr** [comment]
hi Roberto sorry forgot about the topic.
yes i think following this formula or similar would be great.
For the speed i think that the elliptical is sending maybe 0 values in between two different speeds or QZ is picking 0 in between 2 different speeds . I do not know if there is a way to confirm that
Thanks

I am using the elliptical in run mode with zwift ( rouvy do not support running any more) . in this elliptical there is no inclination . Changing the resistance makes it harder so speed and cadence are harder to maintain

**Comment by gewahr** [comment]
Hi Roberto ! concerning my last reply from last month
is there a way to check if my elliptical is sending constant speed to QZ or maybe sending 0 speed in between 2 measurements.
Also is there a way to apply a formula like you mentionned before. Otherwise can speed be adjusted on cadence while including a factor that takes care of the amount of resistance used ( like when I increase the cadence the speed should increase ? also speed should be lower if i increase the resistance but then i am penalised in ZWift then what can be done ?). Again I know you are busy but answer me when you can... Have a nice day

**Reply by cagnulein** [reply]
sorry @gewahr very busy weeks! in this log https://github.com/cagnulein/qdomyos-zwift/files/10909018/debug-lun__mars_6_09_06_02_2023.log the speed sent from the machine was completely unstable.

yes I can do whatever we want :) if you can write me the formula that you would like to add I will add it! Let me know!

**Comment by gewahr** [comment]
Hi Roberto . Thanks for answering in your very busy schedule. I'll start by
sending you debug logs at different resistance levels to check if the speed
is stable as soon as possible while connecting QZ to the elliptical only
without zwift

Le mer. 31 mai 2023 à 10:51, Roberto Viola ***@***.***> a
écrit :

**Reply by cagnulein** [reply]
@gewahr if you are satisfied with the linear equation that i did in the previous comment based on the power, i can add that one to start. let me know

**Comment by gewahr** [comment]
Hi roberto i'm sending you six logs that i did with the elliptical connected to QZ  for 6 resistance levels ( 1, 2,5, 10, 14 and 18). When you can check  if the speed is stable and when you can go with the linear equation that you want and I'll test it . Thanks
[debug-dim__juin_4_10_47_30_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11646654/debug-dim__juin_4_10_47_30_2023.log)
[debug-dim__juin_4_10_49_09_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11646655/debug-dim__juin_4_10_49_09_2023.log)
[debug-dim__juin_4_10_50_26_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11646656/debug-dim__juin_4_10_50_26_2023.log)
[debug-dim__juin_4_10_44_09_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11646657/debug-dim__juin_4_10_44_09_2023.log)
[debug-dim__juin_4_10_45_20_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11646658/debug-dim__juin_4_10_45_20_2023.log)
[debug-dim__juin_4_10_46_21_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11646659/debug-dim__juin_4_10_46_21_2023.log)
   the first log is done with the lower resistance and the following are done with the resistance increasing ...the last one is with the highest resistance R=18.. the machine goes to 22 resistance

**Reply by cagnulein** [reply]
@gewahr checked. In all the files the speed received from the elliptical goes from 9km/h to 12km/h so I guess it's not releated to the resistance level. Is what you are looking for?

**Comment by gewahr** [comment]
No my problem was that sometimes when it is connected to zwift . if i increase the cadence i expect the speed to increase because i am putting more effort and i read for exemple 13km/h then 12km/h then 9 km/h then again 12 in a very short time for approximately the same cadence . It is like zwift  or QZ are receiving a lower value of speed for a split second and speed is not related to the effort that i am applying . Also the resistance in my elliptical can only be changed manually on the console of the elliptical . i would consider that a higher resistance is like a higher incline so my wattage should increase and my speed and cadence drop unless i applying  more effort .. but  i guess zwift in running does not look at that . In conclusion i find that my elliptical connected to zwift does not give a realistic speed related to the cadence or resistance that i apply ....maybe if speed was calculated from cadence .

**Reply by cagnulein** [reply]
I got your point. Yes Zwift for running is using the speed directly without any other variables. So what you're saying makes sense.
So what we can try to do, it's increasing the speed proportionally to the power read from the elliptical itself. What do you think?

**Comment by gewahr** [comment]
I think that would be the best . Because the power generated by the
elliptical increase if I increase the resistance so that simulate a higher
inclination when i run . No emergency..i'm not in a hurry. Thank you very
much !

**Comment by cagnulein** [comment]
ok i will think about how do this!
i will let you know!

Il giorno mar 6 giu 2023 alle 19:38 gewahr ***@***.***> ha
scritto:

> I think that would be the best . Because the power generated by the
> elliptical increase if I increase the resistance so that simulate a higher
> inclination when i run . No emergency..i'm not in a hurry. Thank you very
> much !
>
> Le mar. 6 juin 2023, 10 h 22 a.m., Roberto Viola ***@***.***>
> a écrit :
>
> > I got your point. Yes Zwift for running is using the speed directly
> > without any other variables. So what you're saying makes sense.
> > So what we can try to do, it's increasing the speed proportionally to the
> > power read from the elliptical itself. What do you think?
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/cagnulein/qdomyos-zwift/discussions/1345#discussioncomment-6101505
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AXTA2JPZ4YEXVEG2OKOPTA3XJ44LLANCNFSM6AAAAAAVSL5VJM
> >
> > .
> > You are receiving this because you were mentioned.Message ID:
> > ***@***.***
> > com>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/1345#discussioncomment-6103402>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWGQCPQUIK2XRATOSHLXJ5THZANCNFSM6AAAAAAVSL5VJM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Comment by gewahr** [comment]
tyvm- Let me know if you need more data

**Reply by cagnulein** [reply]
@gewahr I was thinking about this and I have a question: when you increase the resistance on your machine, is the speed on the console matching the speed on qz?

**Comment by gewahr** [comment]
Yes, it does! i just tested it today. ty


Le ven. 16 juin 2023 à 03:58, Roberto Viola ***@***.***> a
écrit :

**Reply by cagnulein** [reply]
ok perfect, i will continute to think how to do it so :)

**Reply by cagnulein** [reply]
ok @gewahr I guess I found a perfect way to do so. I'm reverse the wattage calculation in order to calculate speed. Then you can of course trim the calculation using the speed gain and offset if you are still interested on this. Let me know

**Comment by gewahr** [comment]
Yes, of course  I'm interested. Take your time though; no hurries. Very
much appreciated

thanks
gewahr

Le mar. 11 juil. 2023 à 04:57, Roberto Viola ***@***.***> a
écrit :

**Reply by cagnulein** [reply]
done @gewahr ! it will be in the 2.13.96 (i will send you it later today). Remember to enable the speed from wattage setting in the bike settings (yes it's not in the right place but I reused an exising setting for this). 
Let me know!

**Comment by gewahr** [comment]
THANK YOU ! YOU ARE THE BEST :)

**Comment by gewahr** [comment]
hi roberto!
I have a problem with the last QZ while using my elliptical. when i enable the speed from wattage setting the speed in QZ stays at zero. Also quick question  how is the wattage calculated in QZ for a specific machine ....is there a formula? thank you very much:)

**Comment by cagnulein** [comment]
can you send me a debug log about it?

Il giorno mer 26 lug 2023 alle 16:32 gewahr ***@***.***> ha
scritto:

> Hi Roberto . I have a problem with the latest QZ version while using the
> elliptical : *when i enable the speed from wattage setting* the speed in
> QZ stays at zero
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/1345#discussioncomment-6552407>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWBNXVFFJT7O4PLY3ULXSES77ANCNFSM6AAAAAAVSL5VJM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Reply by gewahr** [reply]
ok i will

**Comment by gewahr** [comment]
in fact only the cadence is showing when i select speed from wattage - the data works only when it is unselected. (some debug file are with the option selected and some are not). thanks
[debug-mer__juill__26_10_38_34_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/12174038/debug-mer__juill__26_10_38_34_2023.log)
[debug-mer__juill__26_10_39_41_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/12174039/debug-mer__juill__26_10_39_41_2023.log)
[debug-mer__juill__26_10_40_25_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/12174040/debug-mer__juill__26_10_40_25_2023.log)

**Comment by cagnulein** [comment]
got the issue, i'm fixing it and I will send you a new one

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 26 lug 2023 alle ore 16:57 gewahr ***@***.***>
ha scritto:

**Comment by gewahr** [comment]
thank you. you are the best:)

**Comment by gewahr** [comment]
hi Roberto ! tried the last build of Qz with my fs5i nordictrack . Datas are back . but something is wrong with the speed. The speed diminishes when i increase my cadence and wattage. So when i go fast i am almost stalled which should be the opposite
[debug-ven__août_11_08_35_29_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/12321708/debug-ven__aout_11_08_35_29_2023.log)

**Comment by cagnulein** [comment]
Right, I calculated pace not the speed with the new formula. I will patch
it and I will send you the new one
Thanks

Il ven 11 ago 2023, 15:05 gewahr ***@***.***> ha scritto:

**Comment by gewahr** [comment]
hi Roberto ! Hope everything is OK. Was trying my elliptical with the last version of QZ and most of the data is not transmitted. Speed stays at 0 and I'm therefore stagnant in Zwift. when you have time have a look at the debug files. thank you very much

[debug-lun__nov__6_08_40_22_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13268310/debug-lun__nov__6_08_40_22_2023.log)
[debug-lun__nov__6_08_40_42_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13268311/debug-lun__nov__6_08_40_42_2023.log)
[debug-lun__nov__6_08_41_30_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13268312/debug-lun__nov__6_08_41_30_2023.log)

**Comment by gewahr** [comment]
have an additional debug files-but it exceeds the 25MB limit. Could send it by smash if you need it. TYVM

**Comment by cagnulein** [comment]
when was the last time you tried?
the last change was for you in july
https://github.com/cagnulein/qdomyos-zwift/commits/master/src/proformelliptical.cpp
after that I didn't change anything.
Did you  update the firmware on the elliptical in the meantime?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 6 nov 2023 alle ore 16:06 gewahr ***@***.***>
ha scritto:

**Comment by gewahr** [comment]
the last time i tried it was maybe 3 weeks ago ...i had the same problem
.No i did not update any firmware as the machine has no connection to the
internet. Only bluetooth

Le lun. 6 nov. 2023 à 10:16, Roberto Viola ***@***.***> a
écrit :

**Comment by gewahr** [comment]
Should I try reinstalling QZ. but it works with all my other devices...only
this one

**Comment by cagnulein** [comment]
@gewahr log checked and I can see the metrics

```
lun. nov. 6 08:41:02 2023 1699278062790 Debug: /Users/cagnulein/qdomyos-zwift/src/bluetooth.cpp void bluetooth::debug(const QString &) "Current Inclination: 0"
lun. nov. 6 08:41:02 2023 1699278062793 Debug: /Users/cagnulein/qdomyos-zwift/src/bluetooth.cpp void bluetooth::debug(const QString &) "Current Cadence: 58"
lun. nov. 6 08:41:02 2023 1699278062795 Debug: /Users/cagnulein/qdomyos-zwift/src/bluetooth.cpp void bluetooth::debug(const QString &) "Current Resistance: 1"
lun. nov. 6 08:41:02 2023 1699278062797 Debug: /Users/cagnulein/qdomyos-zwift/src/bluetooth.cpp void bluetooth::debug(const QString &) "Current Calculate Distance: 0"
lun. nov. 6 08:41:02 2023 1699278062799 Debug: /Users/cagnulein/qdomyos-zwift/src/bluetooth.cpp void bluetooth::debug(const QString &) "Current Watt: 174"
```

what do you see on qz?

**Comment by gewahr** [comment]
I only see wattage  resistance cadence and calories. Elapsed time stays at
0. And of course speed

**Comment by cagnulein** [comment]
got the issue, I will fix this wednesday

for my reference 0x01 0x12

Il giorno lun 6 nov 2023 alle 19:42 gewahr ***@***.***> ha
scritto:

**Reply by gewahr** [reply]
thanks you are the best

**Comment by cagnulein** [comment]
done, i will send you the new beta ASAP

**Comment by cagnulein** [comment]
sent!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 6 nov 2023 alle ore 22:44 gewahr ***@***.***>
ha scritto:

**Comment by gewahr** [comment]
Hi roberto thanks for sending it ..but the results are the same as the previous version. speed is at zero as are all the values that did not work. Nothing changed. is it 2.16.20 ?

**Comment by cagnulein** [comment]
nope, 2.16.22
check test flight

Il giorno gio 9 nov 2023 alle 03:03 gewahr ***@***.***> ha
scritto:

**Comment by gewahr** [comment]
Could you send it to ***@***.*** ? maybe this is why it did
not install. My IPAD is registered to that Email address.
When I open it in testflight it just says open ...doesn't seem to install.
Thanks



Le mer. 8 nov. 2023 à 22:11, Roberto Viola ***@***.***> a
écrit :

**Reply by cagnulein** [reply]
send me an email to roberto.viola83 at gmail.com indicating this ticket , i can't see the email address here :)

**Comment by gewahr** [comment]
Hi Roberto . The new version  2.22 works except that my resistance is reset to 1 after approximately one minute and then the console of the elliptical shows a speed of zero while QZ keeps registering a normal speed for it . Can QZ reset my data or modify data on my console or it just collects data from it ? Here is my debug file in case u need it 
[debug-dim__nov__12_11_26_29_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335618/debug-dim__nov__12_11_26_29_2023.log)
Thanks

**Comment by cagnulein** [comment]
> Can QZ reset my data or modify data on my console or it just collects data from it.

it shouldn't. Does it work right before this version?

The log that you sent me has speed always a 0, maybe the wrong debug log?

**Reply by gewahr** [reply]
Used to do things like that in previous version also. but not all the time

**Reply by gewahr** [reply]
I'll check for the debug file

**Comment by gewahr** [comment]
i have 4 debug files for that day
[debug-dim__nov__12_12_07_18_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335870/debug-dim__nov__12_12_07_18_2023.log)

[debug-dim__nov__12_12_07_30_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335853/debug-di

[debug-dim__nov__12_12_07_30_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335853/debug-dim__nov__12_12_07_30_2023.log)
[debug-dim__nov__12_11_26_29_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335863/debug-dim__nov__12_11_26_29_2023.log)
[debug-dim__nov__12_11_29_35_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335864/debug-dim__nov__12_11_29_35_2023.log)
m__nov__12_12_07_30_2023.log)
[debug-dim__nov__12_11_26_29_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335863/debug-dim__nov__12_11_26_29_2023.log)
[debug-dim__nov__12_11_29_35_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13335864/debug-dim__nov__12_11_29_35_2023.log)

**Comment by cagnulein** [comment]
great I checked and it's the elliptical that it sends the wattage to 0 every minutes for a frame. I'm filtering it and I will send a new beta ASAP

**Comment by gewahr** [comment]
Awesome, thanks!  Could you send it to the gwmultivisions e-mail. Thanks !

**Comment by gewahr** [comment]
Hi Roberto ! just checking if you had time to do  a new beta since we talked about it ( filtering the wattage to o for a frame) . Otherwise, no problem let me know when you have time. Thanks

**Comment by cagnulein** [comment]
not yet, i will try this week. really having bad time to keep the beta here
with all the tickets

Il giorno dom 26 nov 2023 alle 16:54 gewahr ***@***.***> ha
scritto:

**Comment by gewahr** [comment]
Hi Roberto is the last beta 6.26 and if tho i guess it did not yet fixed my problem . Because every minute or so the resistance goes to the minimum and the speeds drops like before. When you have time to fix it give me news. Thanks
[debug-mar__déc__5_18_49_50_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13574200/debug-mar__dec__5_18_49_50_2023.log)
[debug-mar__déc__5_18_58_17_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13574201/debug-mar__dec__5_18_58_17_2023.log)

**Comment by cagnulein** [comment]
ok i tried another approach now, i will send you another one in the next days. it seems that your trainer for some moments, stops to send metrics :(

**Comment by gewahr** [comment]
No worries, thanks for the update!

Le mer. 6 déc. 2023 à 03:28, Roberto Viola ***@***.***> a
écrit :

**Comment by cagnulein** [comment]
sent the update!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 6 dic 2023 alle ore 12:35 gewahr ***@***.***>
ha scritto:
